google/resource_resourcemanager_lien.go

```diff
@@ -0,0 +1,262 @@
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
+package google
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceResourceManagerLien() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceResourceManagerLienCreate,
+		Read:   resourceResourceManagerLienRead,
+		Delete: resourceResourceManagerLienDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceResourceManagerLienImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"reason": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"origin": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"parent": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"restrictions": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceResourceManagerLienCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	reasonProp, err := expandResourceManagerLienReason(d.Get("reason"), d, config)
+	if err != nil {
+		return err
+	}
+	originProp, err := expandResourceManagerLienOrigin(d.Get("origin"), d, config)
+	if err != nil {
+		return err
+	}
+	parentProp, err := expandResourceManagerLienParent(d.Get("parent"), d, config)
+	if err != nil {
+		return err
+	}
+	restrictionsProp, err := expandResourceManagerLienRestrictions(d.Get("restrictions"), d, config)
+	if err != nil {
+		return err
+	}
+
+	obj := map[string]interface{}{
+		"reason":       reasonProp,
+		"origin":       originProp,
+		"parent":       parentProp,
+		"restrictions": restrictionsProp,
+	}
+
+	url, err := replaceVars(d, config, "https://cloudresourcemanager.googleapis.com/v1/liens")
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Creating new Lien: %#v", obj)
+	res, err := Post(config, url, obj)
+	if err != nil {
+		return fmt.Errorf("Error creating Lien: %s", err)
+	}
+
+	// Store the ID now
+	id, err := replaceVars(d, config, "{{name}}")
+	if err != nil {
+		return fmt.Errorf("Error constructing id: %s", err)
+	}
+	d.SetId(id)
+
+	log.Printf("[DEBUG] Finished creating Lien %q: %#v", d.Id(), res)
+
+	d.SetId(res["name"].(string))
```

Comment:
I am not sure why this is required since line 110-114 should basically be doing the same thing right?

Reply:
It _should_ be, except we don't have "name" yet - the only unique ID here is the server-generated one.

Reply:
Can you add a comment in the custom code template to explain this?